src/test/java/org/minions/devfund/abel/WhackAMoleTest.java

```diff
@@ -0,0 +1,186 @@
+    }
+
+    /**
+     * Test WhackAMole place successful.
+     */
+    @Test
+    public void testWhackAMolePlaceTrue() {
+        final int row = 0;
+        final int col = 0;
+        assertTrue(whackAMole.place(row, col));
+    }
+
+    /**
+     * Test WhackAMole place is already taken.
+     */
+    @Test
+    public void testWhackAMolePlaceFalse() {
+        final int row = 0;
+        final int col = 0;
+        whackAMole.place(row, col);
+        assertFalse(whackAMole.place(row, col));
+    }
+
+    /**
+     * Test WhackAMole place is out of Index.
+     */
+    @Test
+    public void testWhackAMolePlaceFalseOutOfIndex() {
+        final int row = 5;
+        final int col = 5;
+        whackAMole.place(row, col);
+        assertFalse(whackAMole.place(row, col));
+    }
+
+    /**
+     * Test WhackAMole place update molesLeft.
+     */
+    @Test
+    public void testWhackAMolePlaceTrueGetMolesLeft() {
+        final int row = 0;
+        final int col = 0;
+        whackAMole.place(row, col);
+        assertEquals(1, whackAMole.getMolesLeft());
+    }
+
+    /**
+     * Test WhackAMole place don't update molesLeft if Out of Index.
+     */
+    @Test
+    public void testWhackAMolePlaceFalseGetMolesLeft() {
+        final int row = -5;
+        final int col = 0;
+        whackAMole.place(row, col);
+        assertEquals(0, whackAMole.getMolesLeft());
+    }
+
+    /**
+     * Test WhackAMole whack a mole successful.
+     */
+    @Test
+    public void testWhackAMoleNoMole() {
+        final int row = 0;
+        final int col = 0;
+        final int expectedAttemptsLeft = 9;
+        whackAMole.whack(row, col);
+        assertEquals(expectedAttemptsLeft, whackAMole.getAttemptsLeft());
+    }
+
+    /**
+     * Test WhackAMole whack a mole successful and update score and molesLeft.
+     */
+    @Test
+    public void testWhackAMoleMoleWhacked() {
+        final int row = 0;
+        final int col = 0;
+        final int expectedAttemptsLeft = 9;
+        final int expectedScore = 1;
+        final int expectedMolesLeft = 0;
+
+        whackAMole.place(row, col);
+        whackAMole.whack(row, col);
+        assertEquals(expectedAttemptsLeft, whackAMole.getAttemptsLeft());
+        assertEquals(expectedScore, whackAMole.getScore());
+        assertEquals(expectedMolesLeft, whackAMole.getMolesLeft());
+    }
+
+    /**
+     * Test WhackAMole whack a mole out of index.
+     */
+    @Test
+    public void testWhackAMoleMoleOutOfIndex() {
+        final int row = 5;
+        final int col = 5;
+        final int expectedAttemptsLeft = 9;
+        whackAMole.whack(row, col);
+        assertEquals(expectedAttemptsLeft, whackAMole.getAttemptsLeft());
+    }
+
+    /**
+     * Test WhackAMole printGridToUser with whacked moles.
+     */
+    @Test
+    public void testWhackAMolePrintGridToUserWithWhackedMoles() {
+        String expected = "W****\n"
+                + "*****\n"
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "*****\n" 12 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)